src/services/Warnings.js

```diff
@@ -58,141 +71,89 @@ const getWarningsZones = (warningsZone, zone, client) => {
-      resptwitter += `#${weatherType} entre as `;
-      resptwitter += `${inicio} e as `;
-      resptwitter += `${fim} para `;
+      strWarning += numPlaces === 1
+        ? 'o distrito de '
+        : 'os distritos de ';
     }
 
-    // Add districts included in warning to both Discord and Twitter message
-    places.forEach(({ local }) => {
-      if (primeiro === 0) {
-        if (['madeira', 'acores'].includes(zone)) {
-          if (places.length === 1 || places.length > 2) {
-            respnovos += `o #${local}`;
-            resptwitter += `o #${local}`;
-          } else if (places.length === 2) {
-            respnovos += `os #${local}`;
-            resptwitter += `os #${local}`;
-          }
-        } else {
-          respnovos += `#${local}`;
-          resptwitter += `#${local}`;
+    if (numPlaces === 1) {
+      const { local } = places[0];
+
+      strWarning += `#${local}`;
+    } else {
+      places.forEach(({ local }, index) => {
+        switch (places.length - index) {
+          case 1:
+            strWarning += `e #${local}`;
+            break;
+          case 2:
+            strWarning += `#${local} `;
+            break;
+          default:
+            strWarning += `#${local}, `;
         }
-      } else if (places.length - 1 === primeiro) {
-        respnovos += `, e #${local}`;
-        resptwitter += `, e #${local}`;
-      } else {
-        respnovos += `, #${local}`;
-        resptwitter += `, #${local}`;
-      }
-      primeiro += 1;
-    });
+      });
+    }
 
     // Add final emojis
-    if (zone === 'continente') {
-      respnovos += ` ${icon} :warning: :information_source:\n\n`;
-      resptwitter += ` ${iconsMap.get(icon)}⚠️ℹ️`;
-    } else if (zone === 'acores') {
-      respnovos += ` dos #Açores ${icon} :warning: :information_source:\n\n`;
-      resptwitter += ` dos #Açores ${iconsMap.get(icon)}⚠️ℹ️`;
+    if (zone === 'acores') {
+      strWarning += ' do arquipélago dos #Açores';
     } else if (zone === 'madeira') {
-      respnovos += ` da #Madeira ${icon} :warning: :information_source:\n\n`;
-      resptwitter += ` da #Madeira ${iconsMap.get(icon)}⚠️ℹ️`;
+      strWarning += ' do arquipélago da #Madeira';
     }
 
     // Send message to Twitter
-    if (clientTwitter && resptwitter !== '') {
-      const fileName = `VOSTPT_aviso_${level}.png`;
-      uploadTweetPhotos(resptwitter, [fileName]);
+    if (clientTwitter && strWarning !== '') {
+      const fileName = `Twitter_Post_Aviso${level}_${removeAccent(weatherType)}.png`;
+
+      const strTwitter = `ℹ️⚠️${iconsMap.get(icon)} ${strWarning} ${iconsMap.get(icon)}⚠️ℹ️`;
+
+      uploadThreadTwitter([{
+        status: strTwitter,
+        media: [fileName],
+      }]);
     }
+
+    strDiscord += `:information_source: :warning: ${icon} ${strWarning} ${icon} :warning: :information_source:\n\n`;
   });
 
   // Send message to Discord
-  if (respnovos !== '') {
+  if (strDiscord !== '') {
```

Comment:
I know this was only a variable renaming thing, but is it really needed to check if empty? Looking at the code below it seems it will have no impact on the outcome.

What's your though?

Reply:
Yes, because if there's no check, the bot will return `***Novos Alertas do(s) <zone> ***` without any warning.
Probably there's a better way to do this, but for now I'll keep as it is and try to check other ways to write this during the weekend.